toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -543,6 +540,16 @@ func customizeOSContents(ctx context.Context, ic *ImageCustomizerParameters) err
 	ctx, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "customize_os_contents")
 	defer span.End()
 
+	resolvedConfig, err := loadBaseConfig(ic.config, ic.configPath)
```

Comment:
The base configs should be loaded early during customization. You probably should move `loadBaseConfig` into `CustomizeImage()`.